core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -240,11 +247,25 @@ private[kafka] object LogValidator extends Logging {
     for (batch <- records.batches.asScala) {
       validateBatch(topicPartition, firstBatch, batch, isFromClient, magic, brokerTopicStats)
 
+      val batchOffsetCounter = new LongRef(0)
```

Comment:
Can we make this a normal long value?

Reply:
Thinking about this more: declaring a `var batchOffsetCounter = 0L` and then incrementing it seem pretty awkward so I changed to `batch.asScala.view.zipWithIndex`